File: web/src/layout/HttpDemoXml.ts
//
// HttpDemoXml.ts
// Created by Khrysalis XML Typescript
//
import { loadHtmlFromString, findViewById, getViewById, replaceViewWithId } from 'butterfly/dist/views/html'
import { customViewSetDelegate } from 'butterfly/dist/views/CustomView'
import htmlForDefault from './http_demo.html'
//! Declares com.lightningkite.butterflytemplate.layouts.HttpDemoXml
export class HttpDemoXml {
    xmlRoot!: HTMLElement;
    progress!: HTMLProgressElement;
    items!: HTMLDivElement;
    loadHtmlString(): string {
        return htmlForDefault;
    }
    setup(dependency: Window): HTMLElement {
        const view = loadHtmlFromString(this.loadHtmlString());
        this.xmlRoot = view
        this.progress = getViewById<HTMLProgressElement>(view, "progress");
        this.items = getViewById<HTMLDivElement>(view, "items");
        return view
    }
}

File: android/src/main/java/com/lightningkite/butterflytemplate/layouts/HttpDemoXml.kt
//
// HttpDemoXml.kt
// Created by Khrysalis XML Android
//
package com.lightningkite.butterflytemplate.layouts

import android.widget.*
import android.view.*
import com.lightningkite.butterfly.views.widget.*
import com.lightningkite.butterfly.views.*
import com.lightningkite.butterfly.android.*
import com.lightningkite.butterflytemplate.R

class HttpDemoXml {

    lateinit var progress: com.lightningkite.butterfly.views.widget.HorizontalProgressBar
    lateinit var items: com.lightningkite.butterfly.views.widget.VerticalRecyclerView
    
    
    lateinit var xmlRoot: View

    fun setup(dependency: ActivityAccess): View {
        val view = LayoutInflater.from(dependency.context).inflate(R.layout.http_demo, null, false)
        return setup(view)
    }
    fun setup(view: View): View {
        xmlRoot = view
        progress = view.findViewById<com.lightningkite.butterfly.views.widget.HorizontalProgressBar>(R.id.progress)
        items = view.findViewById<com.lightningkite.butterfly.views.widget.VerticalRecyclerView>(R.id.items)
        
        
        
        return view
    }
}